### DISCUSSION TITLE
Externalize the authentication process during OIDC4VCI.

### DISCUSSION THREAD

**Discussion by vanhoanHoang** [walt-id/.github#59] [Ideas / Feature Requests]
Dear team,
 
Regarding the OIDC4VCI protocol with Authorization Code flow, the current authentication process is actually mocked. Do you have a plan allowing plugging external authentication mechanisms into the Issuer. For example, user can be well authenticated by other Identity provider and then send the received access token, which should be accepted by Walt Issuer, to the VC endpoint to receive the required VCs.

Thank you
Hoan

**Comment by severinstampler** [comment]
Hi
The issuer portal is indeed meant more as a demo portal, than a ready-made production portal. To make use of the walletkit in a production system, you can either use it as a library in your project, where you can override the API implementation according to your requirements, or you may fork the project and adapt it.
Note, that we are currently re-working our SSI stack, and I think it is a good feature request, that we should consider. 
The walletkit as it is now may be discontinued once the new stack is ready, so we are currently not planning to extend it.
If you want to fork and extend the code, you may submit a pull request, which we will review and merge.
best regards, Severin
